frontend/src/metabase/dashboard/components/DashboardMoveModal.jsx

```diff
@@ -23,11 +35,33 @@ class DashboardMoveModal extends React.Component {
   };
 
   async _moveDashboard() {
+    const { addUndo, createUndo } = this.props;
+    const { selectedCollection } = this.state;
+
     try {
       await DashboardApi.update({
         id: this.props.params.dashboardId,
         collection_id: this.state.selectedCollection.id,
       });
+      addUndo(
+        createUndo({
+          type: "dashboard-move-confirm",
+          message: () => (
+            <Flex align="center">
```

Comment:
It might be nice to pull this out into a component then do `message: () => <DashbaordMoveConfirm collection={selectedCollection} />`

Reply:
Yeah I'll do that. One thing I've found is it's kinda arcane to create these. Knowing to do `addUndo(createUndo(...))` took me a little while to grok initially even though I know it now after making a few of these over the past few months. Ideally it'd be nice to have a `<Toast message={...} {...otherRelevantProps} />` form and an easier way to trigger it. I suppose we could just start with `genToast(message)` or something that uses both under the hood since there should be a fairly limited surface are on what these look like and what they can do.

Reply:
Yeah we could definitely make it easier by abstracting away the Redux stuff, though I don't think it makes a lot of sense to have a declarative style `<Toast ... />` component that requires calling a method on a ref or whatever (like we have with the `*WithTrigger` components)

Maybe a `withToaster` HoC...

```javascript
@withToaster
class MyComponent extends React.Component {
  render() {
    return (
      <button
        onClick={() => this.props.toast(<div>my toast</div>)}
      >
        Toast!
      </button>
    );
  }
}
```